Cargo.toml

```diff
@@ -26,7 +26,7 @@ wabt = "0.9.1"
 wasmer-clif-backend = { path = "lib/clif-backend" }
 wasmer-singlepass-backend = { path = "lib/singlepass-backend", optional = true }
 wasmer-middleware-common = { path = "lib/middleware-common" }
-wasmer-runtime = { path = "lib/runtime" }
+wasmer-runtime = { path = "lib/runtime", default-features = false }
```

Comment:
Hmm, I don't know if we want to change this

Reply:
otherwise, you force the installation of cranelift and it seems to break to compilation

Reply:
Yeah, you can only enable one default backend.
And you cannot disable default features from the `deterministic` feature.

I dealt with this for my metering pr #736 and ended up just documenting that you had to explicitly set the backend with metering on...

`cargo test --features metering,backend-singlepass --no-default-features`

it is a bit heavier to use it, but still configurable and doesn't break anyone else's compile steps. Then again, this is trying to disable default features of a dependency... hmm...

Reply:
See https://github.com/wasmerio/wasmer/pull/736/files#diff-0f742359c7cbd0059a44bc3d831b548c

Reply:
very nice @ethanfrey, thank you!

Reply:
@ethanfrey 
I've reverted that and tried:
```
cargo +nightly build --features=deterministic --no-default-features
```
and for some reason, it installs cranelift...

Reply:
Yeah, what I was thinking was that if a user wanted to do this, then they could pass `default-features = false` in their Cargo.toml of this.

Reply:
@MarkMcCaskey it means that a user will have to use a forked version then?

Reply:
A forked version of what exactly?

It's my understanding that this is a fully configurable way to do this.  In Rust this is done in cargo.toml.  In other things like the go-ext-wasm example, it uses an empty wrapper crate to pass configuration options into the Rust for the dynamic library that it creates.